libtransmission/blocklist.cc

```diff
@@ -49,7 +49,7 @@ void BlocklistFile::ensureLoaded() const
         return;
     }
 
-    auto range = IPv4Range{};
+    auto range = AddressRange{};
```

Comment:
Hmm, what happens when a user runs this for the first time and loads in an old .bin file that expects this file to be full of structs that have a size of `sizeof(IPv4Range)`? Won't we read in a lot of junk here and wind up maybe blocking everything or blocking the wrong ranges?

I'm not sure what the Right Thing is here but this needs some mechanism to force an update to the new binary format. Maybe check the size of the entire file and see if it's a muiltiple of `AddressRange` or not? (I haven't checked, but hopefully `sizeof(AddressRange)` isn't a multiple of `sizeof(IPvRange)`?)

Reply:
Hadn't thought of that. Sadly the IPv4Range is 8 bytes (2 IPv4) and AddressRange is 40 (2 IPv4, 2 IPv6).
I made a check for the file, which should work in all cases but add a tiny bit of overhead on every startup.

Reply:
> Hadn't thought of that. Sadly the IPv4Range is 8 bytes (2 IPv4) and AddressRange is 40 (2 IPv4, 2 IPv6). I made a check for the file, which should work in all cases but adds a tiny bit of overhead on every startup.

All cases except all test cases :).
Fixed now